dll/win32/shell32/shlfolder.cpp

```diff
@@ -274,6 +274,42 @@ HRESULT SHELL32_CompareDetails(IShellFolder2* isf, LPARAM lParam, LPCITEMIDLIST
     return MAKE_COMPARE_HRESULT(ret);
 }
 
+HRESULT SHELL_CompareAllFields(IShellFolder* psf,
+                                 LPCITEMIDLIST pidl1, LPCITEMIDLIST pidl2,
+                                 UINT Count, int Skip)
+{
+    for (UINT i = 0; i < Count; ++i)
+    {
+        if (Skip == (int)i)
+            continue;
+        HRESULT hr = psf->CompareIDs(i, pidl1, pidl2);
+        if (hr && SUCCEEDED(hr))
+            return hr;
+    }
+    return MAKE_COMPARE_HRESULT(0);
+}
+
+HRESULT SHELL_FolderImplCompareIDsTiebreaker(IShellFolder2* psf, LPARAM lParam,
```

Comment:
What do you mean by "tiebreaker" here?

Reply:
The column the caller was interested in resulted in the two items being equal so NT5 will actually [break the tie](https://en.wikipedia.org/wiki/Tiebreaker) by comparing other attributes.

```
LPCITEMIDLIST p1 = SHSimpleIDListFromPath(TEXT("c:\\f1")); // "Filename-only" PIDLs,
LPCITEMIDLIST p2 = SHSimpleIDListFromPath(TEXT("c:\\f2")); //  all other attributes are 0.
IShellFolder*psf; // Leak
PCUITEMID_CHILD child1, child2;
SHBindToParent(p1, IID_PPV_ARG(IShellFolder, &psf), &child1);
SHBindToParent(p2, IID_PPV_ARG(IShellFolder, &psf), &child2);
for (UINT i = 0; i <= 5; ++i)
	printf("%d=%d\n", i, psf->CompareIDs(i, child1, child2));
CopyMemory((char*)child2, child1, child2->mkid.cb); // Make them equal. Will return 0 now meaning it does not compare the PIDL address.
for (UINT i = 0; i <= 5; ++i)
	printf("%d=%d\n", i, psf->CompareIDs(i, child1, child2));
```
![screenshot](https://github.com/user-attachments/assets/f42cdeeb-5cec-4445-b988-995a83873df6)

I'm not a huge fan of this implementation but it is correct for NT5.

At some point this changed, presumably the tie breaking moved to DefView but I have not researched the exact details nor when it changed.

![screenshot](https://github.com/user-attachments/assets/2d3b9619-10be-44a8-b145-e324158d9ecc)

Reply:
I moved the parts of the tie breaking to DefView but retained the strange NT5 handling for NT5 processes.